oauthproxy.go

```diff
@@ -1201,3 +1240,15 @@ func (p *OAuthProxy) errorJSON(rw http.ResponseWriter, code int) {
 	rw.Header().Set("Content-Type", applicationJSON)
 	rw.WriteHeader(code)
 }
+
+// authOnlyRedirectSignIn handles QS 'redirect_signin' parsing on the
+// AuthOnly endpoint to determine redirect to SignInPage/OauthStart
+func authOnlyRedirectSignIn(req *http.Request) bool {
+	query := req.URL.Query()
+	redirectSignIn, err := strconv.ParseBool(query.Get("redirect_signin"))
```

Comment:
Does this potentially make sense as an explicit URL to redirect to in unauthenticated `/oauth2/auth` cases? Gives more flexibility beyond Traefik and eliminates all the signin URL determination logic above.

Mirrors the setup nginx & kubernetes ingress does where they set the explicit signin-url as an annotation.

Reply:
I have switched the redirect to `/oauth2/sign_in` endpoint as the `SignIn` method seems to handle the both SignInPage or OauthStart options depending upon the `SkipProviderButton` value.